src/Microsoft.AspNetCore.Mvc.Core/Internal/ActionDescriptorCollectionProvider.cs

```diff
@@ -43,6 +43,11 @@ public class ActionDescriptorCollectionProvider : IActionDescriptorCollectionPro
 
         private IChangeToken GetCompositeChangeToken()
         {
+            if (_actionDescriptorChangeProviders.Length == 1)
```

Comment:
Just seemed sensible not to have nested `CompositeChangeTokens` in the most common case.